content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-13.md

```diff
@@ -0,0 +1,236 @@
+- `etcd.local.extraArgs.listen-client-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.listen-peer-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.initial-cluster`
+
+    This should be updated to include the hostname and IP address pairs for each control plane node in the cluster. For example:
+
+        "ip-172-31-92-42=https://172.31.92.42:2380,ip-172-31-89-186=https://172.31.89.186:2380,ip-172-31-90-42=https://172.31.90.42:2380"
+
+You must also pass an additional argument (`initial-cluster-state: existing`) to etcd.local.extraArgs.
+
+```shell
+kubectl apply -f kubeadm-config-cm.yaml --force
+```
+
+Start the upgrade:
+
+```shell
+kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+You should see something like the following:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.12.0". Enjoy!
+
+The `kubeadm-config` ConfigMap is now updated from `v1alpha2` version to `v1alpha3`.
+
+### Upgrading additional control plane nodes
+
+Each additional control plane node requires modifications that are different from the first control plane node. Run:
+
+```shell
+kubectl get configmap -n kube-system kubeadm-config -o yaml > kubeadm-config-cm.yaml
+```
+
+Open the file in an editor and replace the following values for `ClusterConfiguration`:
+
+- `etcd.local.extraArgs.advertise-client-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.initial-advertise-peer-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.listen-client-urls`
+
+    This should be updated to the local node's IP address.
+
+- `etcd.local.extraArgs.listen-peer-urls`
+
+    This should be updated to the local node's IP address.
+
+You must also modify the `ClusterStatus` to add a mapping for the current host under apiEndpoints.
+
+Add an annotation for the cri-socket to the current node, for example to use docker:
+
+```shell
+kubectl annotate node <nodename> kubeadm.alpha.kubernetes.io/cri-socket=/var/run/dockershim.sock
+```
+
+Apply the modified kubeadm-config on the node:
+
+```shell
+kubectl apply -f kubeadm-config-cm.yaml --force
+```
+
+Start the upgrade:
+
+```shell
+kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+You should see something like the following:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.12.0". Enjoy!
+
+## External etcd
+
+### Upgrade each control plane
+
+Get a copy of the kubeadm config used to create this cluster. The config should be the same for every node. The config must exist on every control plane node before the upgrade begins.
+
+```
+kubectl get configmap -n kube-system kubeadm-config -o jsonpath={.data.ClusterConfiguration} > kubeadm-config.yaml
+```
+
+Now run the upgrade:
+
+```
+kubeadm upgrade apply v1.13.0 --config kubeadm-config.yaml
+```
+
+Finally run the following command in all other control plane nodes:
```

Comment:
This is confusing, since the heading for this section says Upgrade each control plane, should that heading be updated to indicate it applies to the first control plane instance only, and a separate header added to indicate that this is intended for all subsequent control plane instances?

Reply:
I can do that. Sure.